src/components/NestedForm/NestedFormInput.tsx

```diff
@@ -53,6 +56,8 @@ export class NestedFormInput extends React.PureComponent<data.IGenericFieldInput
 
     private renderEntry(entry: any, index: number) {
         const showDeleteButton = this.props.value.length !== 1;
+        const showIndex = this.props.showIndex !== undefined ? this.props.showIndex : true;
```

Comment:
my thought was to check if showIndex is defined and passed to NestedFormInput, if showIndex is not defined then show index at default.